src/ansys/acp/core/_tree_objects/model.py

```diff
@@ -269,8 +272,20 @@ def _create_stub(self) -> model_pb2_grpc.ObjectServiceStub:
     minimum_analysis_ply_thickness: ReadWriteProperty[float, float] = grpc_data_property(
         "properties.minimum_analysis_ply_thickness"
     )
-    unit_system = grpc_data_property_read_only(
-        "properties.unit_system", from_protobuf=unit_system_type_from_pb
+
+    @staticmethod
+    def _set_unit_system_data_attribute(pb_obj: ObjectInfo, name: str, value: _PROTOBUF_T) -> None:
```

Comment:
Shouldn't we handle that in the backend?

Reply:
It is handled in the backend, in the sense that if the `PUT` request contains _only_ the unit system it will transform the existing `minimum_analysis_ply_thickness`. 

However, if we don't remove the field here, it will transform as (for example):

- start with units: MPA, minimum_analysis_ply_thickness=1e-6 (mm)
- PUT: units: MKS, minimum_analysis_ply_thickness=1e-6 (m)

Since the PUT request can change both the unit system and minimum_analysis_ply_thickness simultaneously, it is interpreted as being internally consistent (i.e., in the new units). 

So, in short: to let the backend do the conversion, we need to ensure we don't send it along in the PUT request.